src/test/java/uk/nhs/prm/repo/ehrtransferservice/database/TransferServiceTest.java

```diff
@@ -0,0 +1,226 @@
+
+        verify(transferRepository).isInboundConversationPresent(inboundConversationId);
+        verify(activityService).captureConversationActivity(inboundConversationId);
+        verify(transferRepository).createConversation(repoIncomingEvent);
+
+        verify(transferRepository, never()).updateConversationStatus(any(), any());
+    }
+
+    @Test
+    void createConversation_ValidRetriedConversationOrResetForRetryRequest_Ok() {
+        // given
+        final UUID inboundConversationId = UUID.randomUUID();
+        final RepoIncomingEvent repoIncomingEvent = createRepoIncomingEvent(inboundConversationId);
+        final ConversationRecord conversationRecord = createConversationRecord(inboundConversationId, INBOUND_REQUEST_SENT);
+
+        // when
+        when(transferRepository.isInboundConversationPresent(inboundConversationId)).thenReturn(true);
+        when(transferRepository.findConversationByInboundConversationId(inboundConversationId)).thenReturn(conversationRecord);
+
+        // then
+        assertDoesNotThrow(() -> transferService.createConversationOrResetForRetry(repoIncomingEvent));
+
+        verify(transferRepository).isInboundConversationPresent(inboundConversationId);
+        verify(transferRepository).findConversationByInboundConversationId(inboundConversationId);
+        verify(activityService).captureConversationActivity(inboundConversationId);
+        verify(transferRepository).updateConversationStatus(inboundConversationId, INBOUND_STARTED);
+
+        verify(transferRepository, never()).createConversation(any());
+    }
+
+    @Test
+    void createExceptionOrResetForRetry() {
+        // given
+        final UUID inboundConversationId = UUID.randomUUID();
+        final RepoIncomingEvent repoIncomingEvent = createRepoIncomingEvent(inboundConversationId);
+        final ConversationRecord conversationRecord = createConversationRecord(inboundConversationId, INBOUND_FAILED);
+
+        // when
+        when(transferRepository.isInboundConversationPresent(inboundConversationId)).thenReturn(true);
+        when(transferRepository.findConversationByInboundConversationId(inboundConversationId)).thenReturn(conversationRecord);
+
+        // then
+        assertThrows(ConversationIneligibleForRetryException.class, () -> transferService.createConversationOrResetForRetry(repoIncomingEvent));
+
+        verify(transferRepository).isInboundConversationPresent(inboundConversationId);
+        verify(transferRepository).findConversationByInboundConversationId(inboundConversationId);
+
+        verify(activityService, never()).captureConversationActivity(any());
+        verify(transferRepository, never()).updateConversationStatus(any(), any());
+        verify(transferRepository, never()).createConversation(any());
+    }
+
+    @Test
+    void updateConversationTransferStatus_ConversationIsTerminated_DoNotUpdateTransferStatusAndConcludeConversation() {

```

Comment:
Terminated?

Reply:
Fair point. Renamed to 'isAlreadyComplete'